jscomp/bsb/bsb_ninja_regen.ml

```diff
@@ -53,26 +60,32 @@ let regenerate_ninja
   | Other _ -> 
     if check_result = Bsb_bsc_version_mismatch then begin 
       Bsb_log.warn "@{<info>Different compiler version@}: clean current repo@.";
-      Bsb_clean.clean_self  per_proj_dir; 
-    end ; 
+      Bsb_clean.clean_self  build_artifacts_dir; 
+    end ;
     
     let config = 
       Bsb_config_parse.interpret_json 
         ~toplevel_package_specs
         ~per_proj_dir in 
+        
     (* create directory, lib/bs, lib/js, lib/es6 etc *)    
-    Bsb_build_util.mkp lib_bs_dir;         
+    Bsb_build_util.mkp build_artifacts_dir;      
+   
     Bsb_package_specs.list_dirs_by config.package_specs
-      (fun x -> 
-        let dir = per_proj_dir // x in (*Unix.EEXIST error*)
+      (fun x ->
+        let dir = build_artifacts_dir // x in (*Unix.EEXIST error*)
         if not (Sys.file_exists dir) then  Unix.mkdir dir 0o777);
-    if toplevel then       
-      Bsb_watcher_gen.generate_sourcedirs_meta
-        ~name:(lib_bs_dir // Literals.sourcedirs_meta)
-        config.file_groups
+    if toplevel then
+      begin
+        Bsb_log.info "@{<info>toplevel@} %s @." (lib_bs_dir // Literals.sourcedirs_meta);
+        Bsb_watcher_gen.generate_sourcedirs_meta
+          ~name:(lib_bs_dir // Literals.sourcedirs_meta)
```

Comment:
this probably needs to be `build_artifacts_dir` as well

Reply:
It already is: `let lib_bs_dir =  build_artifacts_dir // Bsb_config.lib_bs  in`